utils/analytics.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  getAnalytics,
+  getAppInstanceId,
+  logEvent,
+  setAnalyticsCollectionEnabled,
+  logScreenView,
+} from '@react-native-firebase/analytics';
+import { getApp } from '@react-native-firebase/app';
+
+const app = getApp();
+const analytics = getAnalytics(app);
+
+const allowTracking = async () => {
+  const appInstanceId = await getAppInstanceId(analytics);
```

Comment:
There's no error handling, if getAppInstanceId fails, it might lead to app crash.